homeassistant/components/travel_time/__init__.py

```diff
@@ -0,0 +1,193 @@
+    """Track states and offer events for sensors."""
+    component = hass.data[DOMAIN] = EntityComponent(
+        _LOGGER, DOMAIN, hass, SCAN_INTERVAL
+    )
+
+    await component.async_setup(config)
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up a config entry."""
+    return await hass.data[DOMAIN].async_setup_entry(entry)
+
+
+async def async_unload_entry(hass, entry):
+    """Unload a config entry."""
+    return await hass.data[DOMAIN].async_unload_entry(entry)
+
+
+class TravelTimeEntity(Entity):
+    """Representation of a travel_time entity."""
+
+    @property
+    def attribution(self) -> str:
+        """Get the attribution of the travel_time entity."""
+        return None
+
+    @property
+    def destination(self) -> str:
+        """Get the destination coordinates of the travel_time entity."""
+        return None
+
+    @property
+    def destination_name(self) -> str:
+        """Get the destination name of the travel_time entity."""
+        return None
+
+    @property
+    def distance(self) -> str:
+        """Get the distance of the travel_time entity."""
+        return None
+
+    @property
+    def duration(self) -> str:
+        """Get the duration without traffic of the travel_time entity."""
+        return None
+
+    @property
+    def duration_in_traffic(self) -> str:
+        """Get the duration with traffic of the travel_time entity."""
+        return None
+
+    @property
+    def icon(self) -> str:
+        """Icon to use in the frontend."""
+        return None
+
+    @property
+    def mode(self) -> str:
+        """Get the mode of travelling e.g car for this entity."""
+        return None
+
+    @property
+    def name(self) -> str:
+        """Get the name of the travel_time entity."""
+        return None
+
+    @property
+    def origin(self) -> str:
+        """Get the origin coordinates of the travel_time entity."""
+        return None
+
+    @property
+    def origin_name(self) -> str:
+        """Get the origin name of the travel_time entity."""
+        return None
+
+    @property
+    def route(self) -> str:
+        """Get the route of the travel_time entity."""
+        return None
+
+    @property
+    def route_mode(self) -> str:
+        """Get the route_mode e.g fastest of the travel_time entity."""
+        return None
+
+    @property
+    def state(self) -> Optional[str]:
+        """Return the state of the travel_time entity."""
+        return None
+
+    @property
+    def traffic_mode(self) -> Optional[str]:
+        """Return if traffic_mode is enabled for this travel_time entity."""
+        return None
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return the unit this state is expressed in."""
+        return UNIT_OF_MEASUREMENT
+
+    @property
+    def unit_system(self) -> str:
+        """Get the unit system of the travel_time entity."""
+        return self.hass.config.units.name
```

Comment:
Why can't we use unit_of_measurement ?

Reply:
This is used in conjunction with distance in order to indicate what unit system it is measured in. Most integrations allow to define that.
The attribute quickly shows this information.

We could remove it so the user has to remember which unit System he uses or combine the unit and the measurement: "3 m" or "3 y"

Reply:
The user will always remember the unit system it uses ? We should not add things to the backend to solve frontend issues.